### PR TITLE
Some more changes

### PR DESCRIPTION
**Pull request by rsokl**
Fixed a missing plot in the introduction to Jupyter section.

Reformatted the section on IDEs and added a description of PyLance.

Add link to PLYMI's discussion board

Bump copyright year

### CODE REVIEW

File: Python/Module1_GettingStartedWithPython/Getting_Started_With_IDEs_and_Notebooks.md [hunk 1]
```diff
@@ -39,39 +39,49 @@ First and foremost, a good IDE will provide a text editor that will:
...
 **Pros**
 
-- lightweight and elegant
-- completely free
-- works with many different languages, so you only need to familiarize yourself with one IDE if you are a polyglot programmer
-- a huge number of extensions can be downloaded to add functionality to the editor; these are created by a large community of open-source developers.
+- Lightweight and elegant.
+- Completely free.
+- Works with many different languages, so you only need to familiarize yourself with one IDE if you are a polyglot programmer.
+- Offers a huge number of extensions can be downloaded to add functionality to the editor; these are created by a large community of open-source developers.
```
Comment (petarmhg) [inline Python/Module1_GettingStartedWithPython/Getting_Started_With_IDEs_and_Notebooks.md:77]:
```suggestion
- Offers a huge number of extensions that can be downloaded to add functionality to the editor; these are created by a large community of open-source developers.
```

File: Python/Module1_GettingStartedWithPython/Getting_Started_With_IDEs_and_Notebooks.md [hunk 2]
```diff
@@ -39,39 +39,49 @@ First and foremost, a good IDE will provide a text editor that will:
...
 - [has native support for Jupyter notebooks](https://code.visualstudio.com/docs/python/jupyter-support), meaning that you get VSCode's intellisense, debugger, and ability to inspect variables, all in a notebook.
```
Comment (petarmhg) [inline Python/Module1_GettingStartedWithPython/Getting_Started_With_IDEs_and_Notebooks.md:78]:
```suggestion
- [Has native support for Jupyter notebooks](https://code.visualstudio.com/docs/python/jupyter-support), meaning that you get VSCode's intellisense, debugger, and ability to inspect variables, all in a notebook.
```

File: Python/Module1_GettingStartedWithPython/Getting_Started_With_IDEs_and_Notebooks.md [hunk 3]
```diff
@@ -39,39 +39,49 @@ First and foremost, a good IDE will provide a text editor that will:
...
+- Provides incredibly robust [remote coding](https://code.visualstudio.com/docs/remote/remote-overview) and [collaborative coding](https://visualstudio.microsoft.com/services/live-share/) capabilities.
 
 **Cons**
 
-- configuring VSCode for python development can have a moderate learning curve for newcomers
-- many features, like context-aware intellisense and type-inference, are simply more polished and powerful in PyCharm
+- Configuring VSCode for python development can have a moderate learning curve for newcomers.
```
Comment (petarmhg) [inline Python/Module1_GettingStartedWithPython/Getting_Started_With_IDEs_and_Notebooks.md:83]:
```suggestion
- Configuring VSCode for Python development can have a moderate learning curve for newcomers.
```

File: Python/Module1_GettingStartedWithPython/Getting_Started_With_IDEs_and_Notebooks.md [hunk 4]
```diff
@@ -39,39 +39,49 @@ First and foremost, a good IDE will provide a text editor that will:
...
+- Some features, like context-aware intellisense and type-inference, are simply more polished and powerful in PyCharm
```
Comment (petarmhg) [inline Python/Module1_GettingStartedWithPython/Getting_Started_With_IDEs_and_Notebooks.md:84]:
```suggestion
- Some features, like context-aware intellisense and type-inference, are simply more polished and powerful in PyCharm.
```

File: Python/Module1_GettingStartedWithPython/Getting_Started_With_IDEs_and_Notebooks.md [hunk 5]
```diff
@@ -39,39 +39,49 @@ First and foremost, a good IDE will provide a text editor that will:
...
+- Fully-featured, with an excellent debugger, context-dependent "intellisense", type-inference, and more.
+- The free "community version" is extremely robust and feature-rich.
+- Generally provides an extremely high-quality and responsive experience for doing Python development.
 
 **Cons**
 
- - can be resource-heavy, especially for a laptop
- - may be overwhelming to new users (but has good documentation and tutorials)
+ - Can be resource-heavy, especially for a laptop.
+ - May be overwhelming to new users (but has good documentation and tutorials)
```
Comment (petarmhg) [inline Python/Module1_GettingStartedWithPython/Getting_Started_With_IDEs_and_Notebooks.md:61]:
```suggestion
 - May be overwhelming to new users (but has good documentation and tutorials).
```

File: Python/Module1_GettingStartedWithPython/Getting_Started_With_IDEs_and_Notebooks.md [hunk 6]
```diff
@@ -39,39 +39,49 @@ First and foremost, a good IDE will provide a text editor that will:
...
  - Jupyter notebook support requires the premium version of PyCharm, making it inaccessible to newcomers
```
Comment (petarmhg) [inline Python/Module1_GettingStartedWithPython/Getting_Started_With_IDEs_and_Notebooks.md:62]:
```suggestion
 - Jupyter notebook support requires the premium version of PyCharm, making it inaccessible to newcomers.
```

File: Python/Module5_OddsAndEnds/Writing_Good_Code.md [hunk 7]
```diff
@@ -440,7 +440,10 @@ This saves us the trouble of having to run our code, hit an error, read through
...
+Finally, it is also worthwhile to highlight two projects, [mypy](http://mypy-lang.org/) and [pyright](https://github.com/microsoft/pyright), which are used to perform static type-checking on your code based on your type-hints.
+That is, mypy and pyright will both automatically traverse your code and find potential bugs by identifying type conflicts in your code (e.g. trying to capitalize an integer) by checking their annotated and inferred types.
+These tools are especially useful for large-scale code bases. Companies like Dropbox and Microsoft make keen use of static type-checking to identify inconsistencies in their code without having to hit runtime errors. Keep mypy, pyright, and other type-checking utilities in mind as you mature as a Python developer and find yourself working on projects of growing complexity.
+If you are using [VSCode as your IDE](https://www.pythonlikeyoumeanit.com/Module1_GettingStartedWithPython/Getting_Started_With_IDEs_and_Notebooks.html), you can install the [PyLance vscode extension](https://marketplace.visualstudio.com/items?itemName=ms-python.vscode-pylance) to leverage pyright's type checking within your IDE.
```
Comment (petarmhg) [inline Python/Module5_OddsAndEnds/Writing_Good_Code.md:446]:
```suggestion
If you are using [VSCode as your IDE](https://www.pythonlikeyoumeanit.com/Module1_GettingStartedWithPython/Getting_Started_With_IDEs_and_Notebooks.html), you can install the [PyLance VSCode extension](https://marketplace.visualstudio.com/items?itemName=ms-python.vscode-pylance) to leverage pyright's type checking within your IDE.
```

**Review by petarmhg — DISMISSED** [review]
just some minor formatting things

**Comment by rsokl** [comment]
Awesome. Thank you!

**Pull request merged**